src/EA.Weee.RequestHandlers/AatfEvidence/SetNoteStatusHandler.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace EA.Weee.RequestHandlers.AatfEvidence
+{
+    using EA.Prsd.Core;
+    using EA.Prsd.Core.Mapper;
+    using EA.Weee.DataAccess;
+    using EA.Weee.DataAccess.DataAccess;
+    using EA.Weee.RequestHandlers.Security;
+    using EA.Weee.Requests.Note;
+    using Prsd.Core.Mediator;
+    using System;
+    using System.Threading.Tasks;
+
+    public class SetNoteStatusHandler : IRequestHandler<SetNoteStatus, Guid>
+    {
+        private readonly IWeeeAuthorization authorization;
+        private readonly IEvidenceDataAccess evidenceDataAccess;
+        private readonly IMapper mapper;
+        private readonly WeeeContext context;
+
+        public SetNoteStatusHandler(WeeeContext context, IWeeeAuthorization authorization,
+            IEvidenceDataAccess evidenceDataAccess,
+            IMapper mapper)
+        {
+            this.authorization = authorization;
+            this.evidenceDataAccess = evidenceDataAccess;
+            this.mapper = mapper;
+            this.context = context;
+        }
+
+        public async Task<Guid> HandleAsync(SetNoteStatus message)
+        {
+            authorization.EnsureCanAccessExternalArea();
```

Comment:
Add a check for scheme access similar to how its happened in GetEvidenceNoteForSchemeRequest

Reply:
check added

Reply:
Need unit and integration tests for the handler.

Reply:
scheme access ensured - using EnsureSchemeAccess()